crates/gossip/src/gossip.rs

```diff
@@ -1040,10 +1041,8 @@ mod test {
         )
         .await;
 
-        // The second harness has an empty arc set, so it will be lower priority for target
-        // selection in gossip initiate.
         let harness_2 = factory.new_instance().await;
-        let agent_2 = harness_2.join_local_agent(DhtArc::Empty).await;
```

Comment:
This was a trick to make the test work previously, but the selection logic changing means it no longer works. Instead, this agent is only added to the peer store later, after gossip rounds with the unavailable agents has been attempted.